### PR TITLE
GTEST/JENKINS: Fix CUDA testing 

### PR DESCRIPTION
**Pull request by bureddy**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9548/ for details (Mellanox internal link).

**Comment by bureddy** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9598/ for details (Mellanox internal link).

**Comment by bureddy** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9605/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9622/ for details (Mellanox internal link).

File: src/ucs/async/async.c [hunk 1]
```diff
@@ -615,7 +615,13 @@ void ucs_async_poll(ucs_async_context_t *async)
 
 void ucs_async_global_init()
 {
-    pthread_rwlock_init(&ucs_async_global_context.handlers_lock, NULL);
+    int ret;
+
+    ret = pthread_rwlock_init(&ucs_async_global_context.handlers_lock, NULL);
```
Comment (yosefe) [inline src/ucs/async/async.c]: weird, any idea why this wasn't enough:
```
static ucs_async_global_context_t ucs_async_global_context = {
    .handlers_lock   = PTHREAD_RWLOCK_INITIALIZER,
```

Comment (yosefe) [reply src/ucs/async/async.c]: never mind, i got it

**Pull request merged**